data/transactions/logic/fields_test.go

```diff
@@ -270,18 +270,18 @@ func TestAcctParamsFieldsVersions(t *testing.T) {
 	t.Parallel()
 
 	for _, field := range acctParamsFieldSpecs {
-		text := fmt.Sprintf("txn Sender; acct_params_get %s; assert;", field.field.String())
+		text := fmt.Sprintf("txn Sender; acct_params_get %s; assert;", field.field)
 		if field.ftype == StackBytes {
 			text += "global ZeroAddress; concat; len" // use concat to prove we have bytes
 		} else {
 			text += "global ZeroAddress; len; +" // use + to prove we have an int
 		}
-		// check assembler fails if version before introduction
-		for v := uint64(2); v <= AssemblerMaxVersion; v++ {
-			ep, txn, ledger := makeSampleEnv()
+
+		testLogicRange(t, 4, 0, func(t *testing.T, ep *EvalParams, txn *transactions.Transaction, ledger *Ledger) {
```

Comment:
It's an interesting philosophical question.

On the one hand, it seems appropriate to use constants when they exist, to make tests like this clearer.  On the other hand, part of the point, with some some of these tests, is to ensure we don't change anything that changes the way old code works.  If we used constants like this in our tests, then when someone changes the constant, the tests would change too, and nothing would break! But such a change is very bad, since it would change the past.

In this _particular_ case, the constant chosen is kind of pointless. It could be anything from 2 on up, because the code in the for loop decides what error to expect. I think `directRefEnabledVersion` here would be more confusing than helpful, since the test is about whether the acct params field exists in a particular version, which is unaffected by whether direct refs are enabled.

Reply:
> I think directRefEnabledVersion here would be more confusing than helpful, since the test is about whether the acct params field exists in a particular version, which is unaffected by whether direct refs are enabled.

I see, thanks for explaining! resolving.